File: frontend-mouse-kerramientas/types/rating.ts
export interface Rating {
  id: number;
  tool_id: number;
  user_id: number;
  rating: number;
  comment?: string;
  created_at: string;
  updated_at?: string;
}

export interface RatingWithUser extends Rating {
  user_username: string;
  user_full_name?: string;
}

export interface RatingWithTool extends Rating {
  tool_name: string;
  tool_brand: string;
  tool_model: string;
}

export interface RatingCreate {
  tool_id: number;
  rating: number;
  comment?: string;
}

export interface RatingUpdate {
  rating?: number;
  comment?: string;
}

export interface RatingStats {
  total_ratings: number;
  average_rating: number;
  rating_distribution: {
    1: number;
    2: number;
    3: number;
    4: number;
    5: number;
  };
}

File: backend-mouse-kerramientas/app/schemas/rating.py
"""
Esquemas Pydantic para validación de datos de calificaciones.
"""
from typing import Optional
from pydantic import BaseModel, Field
from datetime import datetime


class RatingBase(BaseModel):
    """Esquema base para calificaciones."""
    rating: float = Field(..., ge=1.0, le=5.0, description="Calificación de 1 a 5 estrellas")
    comment: Optional[str] = Field(None, max_length=500, description="Comentario opcional")


class RatingCreate(RatingBase):
    """Esquema para crear una nueva calificación."""
    tool_id: int = Field(..., description="ID de la herramienta a calificar")


class RatingUpdate(BaseModel):
    """Esquema para actualizar una calificación existente."""
    rating: Optional[float] = Field(None, ge=1.0, le=5.0, description="Calificación de 1 a 5 estrellas")
    comment: Optional[str] = Field(None, max_length=500, description="Comentario opcional")


class Rating(RatingBase):
    """Esquema para respuestas de calificación."""
    id: int
    tool_id: int
    user_id: int
    created_at: datetime
    updated_at: Optional[datetime]

    class Config:
        """Configuración del modelo."""
        orm_mode = True


class RatingWithUser(Rating):
    """Esquema para calificación con información del usuario."""
    user_username: str
    user_full_name: Optional[str]


class RatingWithTool(Rating):
    """Esquema para calificación con información de la herramienta."""
    tool_name: str
    tool_brand: str
    tool_model: str


class RatingStats(BaseModel):
    """Esquema para estadísticas de calificaciones."""
    total_ratings: int
    average_rating: float
    rating_distribution: dict
